index.js

```diff
@@ -1,4 +1,39 @@
-export default function uniqueRandom(minimum, maximum) {
+function _exhaustive(minimum, maximum) {
+	function integersInRange(minimum, maximum) {
+		const array = [];
+		for (let index = minimum; index < maximum + 1; index++) {
+			array.push(index);
+		}
+
+		return array;
+	}
+
+	const defaultArray = integersInRange(minimum, maximum);
+	let toExtract = [...defaultArray];
+	let previousValue;
+
+	return function random() {
+		if (toExtract.length === 0) {
+			toExtract = [...defaultArray];
```

Comment:
Either:

- Have `integersInRange` return an iterable so there are never 2 arrays stored in memory
- or move values from one array to the other, swapping references when empty

Reply:
Just updated with reference swap:
every time there a number is extracted, it's stored in `restorer` and deleted from `toExtract`, when empty, swapping `toExtract` with `restorer`.